src/mmap.rs

```diff
@@ -447,14 +447,23 @@ impl GuestMemory for GuestMemoryMmap {
         self.regions.len()
     }
 
-    fn find_region(&self, addr: GuestAddress) -> Option<&GuestRegionMmap> {
-        for region in self.regions.iter() {
-            if addr >= region.start_addr() && addr <= region.end_addr() {
-                return Some(region);
-            }
-        }
-        None
-    }
+    fn find_region(&self, addr: GuestAddress) -> Option<&GuestRegionMmap> {           
+        let bs_region = self.regions.binary_search_by_key(&addr, |x| x.start_addr()); 
```

Comment:
This looks like it's changing the behavior of `find_region`. The function was previously returning the region that contains the `addr` passed as a parameter. Now it looks like it returns the region that starts with `addr`. Is this true?

Reply:
Return value is same with previous version. In previous version it just does iterate through regions. Now it returns just region with index (e.g regions[x]) which founded with BS.

Reply:
It does same thing still. In BS version, it searches exact match (starts with `addr`) if can't find exact value that starts with `addr` then searches nearest value (check `end_addr` in like previous version) in line 458.

Reply:
Ok, thanks. Let me check that again.